vision/dcgan_mnist/dcgan_mnist.jl

```diff
@@ -56,12 +56,12 @@ end
 
 # Loss functions
 function discriminator_loss(real_output, fake_output)
-    real_loss = mean(logitbinarycrossentropy.(real_output, 1f0))
-    fake_loss = mean(logitbinarycrossentropy.(fake_output, 0f0))
+    real_loss = mean(logitbinarycrossentropy(real_output, 1f0))
+    fake_loss = mean(logitbinarycrossentropy(fake_output, 0f0))
```

Comment:
As a best practice, we should use `one(eltype(real_output))` and `zero(eltype(real_output))`, etc. Not a big deal for this function, but new users often get bit by `Float32`/`Float64` mismatch elsewhere, so we should make our sample code illustrate how to always match the argument types.

Reply:
```suggestion
    real_loss = logitbinarycrossentropy(real_output, 1)
    fake_loss = logitbinarycrossentropy(fake_output, 0)
```
All losses now perform `mean` reduction by default. Integers are also fine in this case, otherwise @darsnack could be shortened a bit with `one(real_output)`

Reply:
Fixed these, but the suggested `one(real_output)` does not work for me (`one` is only implemented for scalars or square matrices).